src/main/java/macrobase/transform/FeatureTransform.java

```diff
@@ -0,0 +1,39 @@
+package macrobase.transform;
+
+import macrobase.conf.MacroBaseConf;
+import macrobase.datamodel.Datum;
+import macrobase.ingest.DataIngester;
+
+import javax.xml.crypto.Data;
+import java.util.Iterator;
+import java.util.function.Consumer;
+
+public class FeatureTransform implements Iterator<Datum> {
+
+    DataIngester ingester;
+
+    public FeatureTransform(MacroBaseConf conf, DataIngester ingester) {
+        // TODO: apply specified configurations
+        this.ingester = ingester;
+    }
+
+    /**
+      * Applies the transformation this class represents to the datum given and
+      * outputs a new transformed datum.
+      * @param datumToTransform datum that should be transformed
+      * @return new Datum that has transformation applied to it.
+      */
+    protected Datum transform(Datum datumToTransform) {
```

Comment:
According to http://stackoverflow.com/questions/3851747/java-how-to-implement-private-abstract-methods you can have `protected abstract` but not `private abstract`.